doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,255 @@
+                          “lane1” : “7”,
+                          “lane2” : “7”,
+                          “lane3” : “7”,
+                          “lane4” : “7”,
+                          “lane5” : “7”,
+                          “lane6” : “7”,
+                          “lane7” : “7” 
+                      } 
+                  } 
+              } 
+          } 
+      }, 
+      “PORT_MEDIA_SETTINGS” : { 
+          “32”: { 
+              “EOPTOLINK”: {
+                  "Default": {
+                      “SI_PARAM_RX_OUTPUT_PRE” : { 
+                          “lane0” : “9”,
+                          “lane1” : “9”,
+                          “lane2” : “9”,
+                          “lane3” : “9”,
+                          “lane4” : “9”,
+                          “lane5” : “9”,
+                          “lane6” : “9”,
+                          “lane7” : “9”,
+                      } 
+                  } 
+              } 
+          } 
+      }
+  }
+```
+
+# Proposed Work-Flow
+Please refer below points in line with flow diagram.
+
+1. When CMIS-supported module insertion happens in XCVRD, the module will progress to the AP_CONFIG state in the CMIS state machine. After which, when the module is in DataPathDeactivated or Disabled state, check if the OPTICS_SI_PARAM.json file is parsed successfully and if lane speed needs special Signal Integrity (SI) settings.
+
+2. If both of the above conditions are met, then proceed to generate the key (module key) and retrieve the SI attribute list.  
+  2.1. The JSON file has two directions (TX_SETTING and RX_SETTING) blocks. Each of these blocks contains sub-blocks of module vendors and other details that will help identify the best match to generate the SI attribute list, if applicable.  
+  2.2.  The EQ_FIXED flag is required to validate if the TX input EQ (equalization) setting is adaptive or fixed. TX EQ settings are adaptive by default and should be disabled to apply host-defined SI settings. Currently, fixed EQ settings are not recommended for TX input EQ settings. This kind of validation is not required to apply RX EQ settings. If EQ_FIXED is set to true only, then flow will move ahead with list generation, or else it will return an empty list for the direction (TX/RX) setting (2.10).   
+  2.3. Generate module key based on Data Path(DP) lane speed + module vendor name, + module vendor part number.    
+  2.4. Using this module key, the search begins for the detected port in the GLOBAL_MEDIA_SETTINGS section of the applicable direction setting block. If the module key matches entries or any search that matches for (2.5) to (2.8) sections, then SI attributes from this section are copied to the SI param attribute list (2.9).   
+  2.5. If no match happens in 2.4, reduce the search to module default + speed of the detected port in GLOBAL_MEDIA_SETTINGS  
+  2.6. If no match happens in 2.5, reduce the search to the speed of the detected port in GLOBAL_MEDIA_SETTINGS  
```

Comment:
please clarify the difference in search in 2.5 and 2.6. Which "module default" is meant in 2.5 which is omitted in 2.6 ?

Reply:
Section 2.5 discusses the module default and speed validation. For example, if a specific module vendor requires signal integrity (SI) settings for a 100G lane speed, regardless of the module vendor's part number (i.e., module default), this section will include those values. Even if a module vendor has hundreds of part numbers, if it has default values for a particular speed, they can be found here.

On the other hand, Section 2.6 addresses cases where the platform requires different SI settings for a given lane speed on identified ports, regardless of the module being used. For instance, if a certain platform needs SI settings for a range of ports operating at 100G lane speed, irrespective of the module installed, this section handles those requirements.

Reply:
So if I understand correctly - 2.5 will search (lane speed) + (module vendor name) and 2.6 will search (lane speed) + (port number)?

Reply:
Yes. All the checks are based on port number. 2.5 will search (lane speed) + (module vendor name) for detected port.
2.6 will search (lane speed) for detected port.

Reply:
Is it possible to update 2.4 and 2.5 to use the same field naming like 2.3?

Reply:
Sorry. Could you please provide more details or what you are referring to as field name? I will try to fix it. Thanks